library/Freckle/App/Kafka/Consumer.hs

```diff
@@ -0,0 +1,166 @@
+{-# LANGUAGE ApplicativeDo #-}
+
+module Freckle.App.Kafka.Consumer
+  ( HasKafkaConsumer (..)
+  , withKafkaConsumer
+  , KafkaConsumerConfig (..)
+  , envKafkaConsumerConfig
+  , runConsumer
+  ) where
+
+import Freckle.App.Prelude
+
+import Blammo.Logging
+import qualified Control.Immortal as Immortal
+import Control.Lens (Lens', view)
+import Data.Aeson
+import qualified Data.List.NonEmpty as NE
+import qualified Data.Map.Strict as Map
+import qualified Data.Text as T
+import qualified Env
+import Freckle.App.Env
+import Freckle.App.Kafka (envKafkaBrokerAddresses)
+import Kafka.Consumer hiding
+  ( Timeout
+  , closeConsumer
+  , newConsumer
+  , runConsumer
+  , subscription
+  )
+import qualified Kafka.Consumer as Kafka
+import UnliftIO.Exception (bracket, displayException, throwIO)
+
+data KafkaConsumerConfig = KafkaConsumerConfig
+  { kafkaConsumerConfigBrokerAddresses :: NonEmpty BrokerAddress
+  -- ^ The list of host/port pairs for establishing the initial connection
+  -- to the Kafka cluster.
+  --
+  -- This is the `bootstrap.servers` Kafka consumer configuration property.
+  , kafkaConsumerConfigGroupId :: ConsumerGroupId
+  -- ^ The consumer group id to which the consumer belongs.
+  --
+  -- This is the `group.id` Kafka consumer configuration property.
+  , kafkaConsumerConfigTopics :: NonEmpty TopicName
+  -- ^ The list of topic names that for which the consumer will subscribe.
+  , kafkaConsumerConfigOffsetReset :: OffsetReset
+  -- ^ The offset reset parameter used when there is no initial offset in Kafka.
+  --
+  -- This is the `auto.offset.reset` Kafka consumer configuration property.
+  , kafkaConsumerConfigExtraSubscriptionProps :: Map Text Text
```

Comment:
I am assuming this is to allow configuration of the underlying `Subscription`? Maybe a mention of `Kafka.Consumer` would be nice here.

Reply:
Kind of; `hw-kafka-client` has no documentation on these props -- you have to use the Kafka Consumer documentation instead. I was hoping this is legible by other record properties above which reference `bootstrap.servers`, `group.id`, and `auto.offset.reset`.